### ISSUE TITLE
Broken highlighting when having PHP code as JS function parameters

### ISSUE THREAD

**Issue by antaed** [pangloss/vim-javascript#1094]
A typical example in a PHP file:

```
<button onclick="addToCart(<?php echo $id; ?>, <?php echo $price; ?>, <?php echo $quantity; ?>)">
   Add to cart 
</button>
```

The PHP delimiters seem to be misinterpreted as JavaScript operators in this case.
The issue doesn't occur with Vim's default syntax highlighting.
Is there something I should do to fix this? Thanks

**Comment by bounceme** [comment]
so the php file with tries to use Javascript for that? if so the bug is with the php syntax. or are you doing some kind of jsx<html<php thing?

**Comment by antaed** [comment]
I don't use JSX. I had to change the title because the issue more generally occurs in any PHP>JS>PHP scenario. Another example:
`<script>
var _prices = <?php echo $prices ? json_encode($prices) : 'false';?>
</script>`
The same issue here, '<?' and '?>' are confused as JS operators and highlighting breaks from there.
It might be a PHP syntax bug but it only occured after installing this plugin, so I'm guessing there could be workaround to solve it.

**Comment by amadeus** [comment]
Yeah so the only way to fix this would be via our plugin system but that means someone would need to write the plugin for this. I don’t really use PHP and therefore don’t really have an interest in building this. If someone would like to take it on, I am happy to review PRs tho.

**Comment by antaed** [comment]
Is there a command I could use to toggle the plugin on/off? This would make my life easier until somebody hopefully comes up with a fix.

**Comment by antaed** [comment]
Thanks but it won't help, the file type is already set to php...

**Comment by bounceme** [comment]
this is a far fetched suggestion: 
````
    <script><!--
    var _prices = //-->
    <?php echo $prices ? json_encode($prices) : 'false';?> </script>
````

**Comment by antaed** [comment]
Yeah, this is very similar to what I came up with. It works but it's not very convenient...
